pkg/cloudprovider/fake/cloudprovider.go

```diff
@@ -158,6 +159,9 @@ func (c *CloudProvider) Get(_ context.Context, id string) (*v1beta1.NodeClaim, e
 	defer c.mu.RUnlock()
 
 	if nodeClaim, ok := c.CreatedNodeClaims[id]; ok {
+		if nodeClaim.StatusConditions().GetCondition(v1beta1.Terminating).IsTrue() {
```

Comment:
Why do we need this check here now?

Reply:
Had added this here when we were relying on cloudProvider.Get() to check instance termination status. Forgot to revert it. Reverting it now.